elemental-system-agent (seems to be) started before timesyncd

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

When booting an installed image on Raspberry Pi, `elemental-system-agent` complains about 
```
panic: error while connecting to Kubernetes cluster: Get "https://192.168.0.28/k8s/clusters/local/version": x509: certificate has expired or is not yet valid: current time 2022-11-03T12:01:01Z is before 2022-12-01T14:19:13Z
```

after a while these messages stop as `systemd-timesyncd` kicks in.

Raspberry Pi has no (buffered) hardware clock and depends on network and timesyncd to get the correct date/time.

**What did you expect to happen:**

`elemental-system-agent` not getting in panic.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Elemental release version (use `cat /etc/os-release`):
- Rancher version:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:

--- a/Dockerfile.image
+++ b/Dockerfile.image
@@ -59,7 +59,7 @@ COPY --from=elemental-cli /usr/bin/elemental /usr/bin/elemental
 COPY framework/files/ /
 
 # Enable services
-RUN systemctl enable NetworkManager sshd elemental-populate-node-labels 
+RUN systemctl enable NetworkManager sshd elemental-populate-node-labels systemd-timesyncd
 
 ARG IMAGE_TAG=latest
 ARG IMAGE_COMMIT=""

--- a/framework/files/usr/lib/systemd/system/elemental-system-agent.service
+++ b/framework/files/usr/lib/systemd/system/elemental-system-agent.service
@@ -3,6 +3,7 @@ Description=Elemental System Agent
 Documentation=https://github.com/rancher/system-agent
 Wants=network-online.target
 After=network-online.target
+After=time-sync.target
 ConditionPathExists=!/run/cos/live_mode
 ConditionPathExists=!/etc/systemd/system/rancher-system-agent.service
 

